src/main/java/hudson/plugins/translation/L10nDecorator.java

```diff
@@ -97,9 +100,15 @@ public String encodeRecording(StaplerRequest request) throws IOException {
      * Does the opposite of {@link #encodeRecording(StaplerRequest)}.
      */
     public List<Msg> decode(StaplerRequest request) throws IOException {
-        BufferedReader r = new BufferedReader(new InputStreamReader(new GZIPInputStream(new CipherInputStream(
+        final GZIPInputStream gzipInpurStream = new GZIPInputStream(new CipherInputStream(
                 new ByteArrayInputStream(Base64.decode(request.getParameter("bundles").toCharArray())),
-                getCipher(DECRYPT_MODE)))));
+                getCipher(DECRYPT_MODE)));  
+        final BufferedReader r;
+        try {
+            r = new BufferedReader(new InputStreamReader(gzipInpurStream, "UTF-8"));
+        } catch (UnsupportedEncodingException ex) {
+            throw new IOException("UTF-8 encoding is not supported");
```

Comment:
I'd rather disagree. The user will see the issue in any case, but throwing of a known exception reduces the possible impact and prevents Jenkins from blowing up on RuntimeException like it does sometimes. I would just a TODO for the replacement when JDK7 becomes a baseline.